src/ocm/components/clusterConfiguration/DiscoveryImageForm.tsx

```diff
@@ -50,7 +50,31 @@ const DiscoveryImageForm = ({
   ) => {
     if (cluster.id && infraEnv?.id) {
       if (formValues['imageType'] === 'discovery-image-ipxe') {
```

Comment:
there is quite a lot of duplication in if/else blocks. How about replacing them with
```ts
const isIPXE = formValues['imageType'] === 'discovery-image-ipxe';
      try {
        const { updatedCluster } = await DiscoveryImageFormService.update(
          cluster.id,
          cluster.tags,
          infraEnv.id,
          formValues,
          undefined,
          isIPXE,
        );
        await (isIPXE ? onSuccessIpxe() : onSuccess());
        dispatch(updateCluster(updatedCluster));
      } catch (error) {
        handleApiError(error, () => {
          formikActions.setStatus({
            error: {
              title: isIPXE ? 'Failed to create ipxe scripts' : 'Failed to create discovery image',
              message: getApiErrorMessage(error),
            },
          });
        });
        if (isUnknownServerError(error as Error)) {
          dispatch(setServerUpdateError());
          onCancel();
        }
      }
```